cmd/chisel/cmd_cut.go

```diff
@@ -99,14 +99,57 @@ func (cmd *cmdCut) Execute(args []string) error {
 		archives[archiveName] = openArchive
 	}
 
+	pkgArchives, err := selectPkgArchives(archives, selection)
+	if err != nil {
+		return err
+	}
+
 	_, err = slicer.Run(&slicer.RunOptions{
-		Selection: selection,
-		Archives:  archives,
-		TargetDir: cmd.RootDir,
+		Selection:   selection,
+		PkgArchives: pkgArchives,
+		TargetDir:   cmd.RootDir,
 	})
 	return err
 }
 
+// selectPkgArchives selects the appropriate archive for each selected slice
+// package. It returns a map of archives indexed by package names.
+func selectPkgArchives(archives map[string]archive.Archive, selection *setup.Selection) (map[string]archive.Archive, error) {
+	pkgArchives := make(map[string]archive.Archive)
+	for _, s := range selection.Slices {
+		pkg := selection.Release.Packages[s.Package]
+		if _, ok := pkgArchives[pkg.Name]; ok {
+			continue
+		}
+		if pkg.Archive == "" {
+			var chosen *setup.Archive
+			for _, releaseArchive := range selection.Release.Archives {
+				archive := archives[releaseArchive.Name]
+				if archive == nil || !archive.Exists(pkg.Name) {
+					continue
+				}
+				if chosen == nil || chosen.Priority < releaseArchive.Priority {
+					chosen = releaseArchive
+				}
```

Comment:
You can probably sort `selection.Release.Archives` by priority (i.e. `slices.SortFunc`) and then go in order from first to last instead of having to find the chosen one each time and having to `archive.Exists` them.

Reply:
Nice, thanks! Did consider it once before. But I reckon the `archive.Exists` is the heavier operation here.